### DISCUSSION TITLE
different table_settings for tables on the same page

### DISCUSSION THREAD

**Discussion by erindavisdataiku** [jsvine/pdfplumber#753] [Ask for help with specific PDFs]
Hi there,
I have multiple tables on a single pdf page and each table has different needs for table_settings.  What I attempted to do was use extract(table_settings) on each table in my tables object (tables = AE_p2.find_tables() ), but I don't think I can use table_settings in the extract method.  What's the best way to use different table_settings for the individual tables on each page.

**Comment by jsvine** [comment]
Hi @erindavisdataiku, and fun/tricky topic! I'd suggest something like this:

```python
tables = page.find_tables(finder_settings)
e0 = page.crop(tables[0].bbox).extract_table(e0_settings)
e1 = page.crop(tables[1].bbox).extract_table(e1_settings)
```

... which could be refactored depending on how exactly you're determining which extraction settings apply to which tables.

Does that help?

**Comment by erindavisdataiku** [comment]
I think so, what is supposed to go in finder_settings? I'm assuming en_settings are the table_settings for each individual table?

**Reply by jsvine** [reply]
`finder_settings` would be whatever table settings (the same kind of settings you use for extraction) that you planned to use for identifying the tables. But perhaps I'm misunderstanding the particular question/difficulty here.

**Reply by erindavisdataiku** [reply]
Hi jsvine,
Sorry if I'm being obtuse, quite new to pdfplumber.  Each table will have to have it's own explicit horizontal and vertical lines so I'm assuming that's what will go in e0_settings and e1_settings.  Just curious what else you think needs to be specified for finding the tables in the screenshot above.

**Reply by jsvine** [reply]
> Each table will have to have it's own explicit horizontal and vertical lines so I'm assuming that's what will go in e0_settings and e1_settings.

Yep, sounds reasonable to me!

A bit hard to anticipate other issues/aspects, however, without having the PDF on hand — behind the scenes, two similar-looking PDFs can have very different internal structures.
